packages/komencikit/src/kit.ts

```diff
@@ -0,0 +1,424 @@
+  GetDistributedBlindedPepperResp,
+  requestSubsidisedAttestations,
+  startSession,
+  StartSessionPayload,
+  submitMetaTransaction,
+} from './actions'
+import { KomenciClient } from './client'
+import {
+  AuthenticationFailed,
+  FetchError,
+  FetchErrorTypes,
+  InvalidWallet,
+  KomenciKitErrorTypes,
+  LoginSignatureError,
+  TxError,
+  TxErrorTypes,
+  TxEventNotFound,
+  TxRevertError,
+  TxTimeoutError,
+} from './errors'
+import { buildLoginTypedData } from './login'
+import { retry } from './retry'
+import { verifyWallet } from './verifyWallet'
+
+const TAG = 'KomenciKit'
+
+interface KomenciOptions {
+  url: string
+  token?: string
+  txRetryTimeoutMs: number
+  txPollingIntervalMs: number
+}
+
+const DEFAULT_OPTIONS: Pick<KomenciOptions, 'txRetryTimeoutMs' | 'txPollingIntervalMs'> = {
+  txRetryTimeoutMs: 20000,
+  txPollingIntervalMs: 100,
+}
+
+export type KomenciOptionsInput = Omit<KomenciOptions, keyof typeof DEFAULT_OPTIONS> &
+  Partial<KomenciOptions>
+
+export class KomenciKit {
+  private client: KomenciClient
+  private options: KomenciOptions
+  private externalAccount: string
+
+  constructor(
+    private contractKit: ContractKit,
+    externalAccount: string,
+    options: KomenciOptionsInput
+  ) {
+    this.externalAccount = normalizeAddressWith0x(externalAccount)
+    this.options = {
+      ...DEFAULT_OPTIONS,
+      ...options,
+    }
+
+    this.client = new KomenciClient(this.options.url, this.options.token)
+  }
+
+  /**
+   * startSession: uses the /v1/startSession endpoint to start a Komenci session
+   * It results in a token that is saved in the client automatically and
+   * will be used on subsequent requests.
+   *
+   * @param captchaToken - an unspent captcha token
+   * @return Result<token, error>
+   */
+  startSession = async (
+    captchaToken: string
+  ): Promise<Result<string, FetchError | AuthenticationFailed | LoginSignatureError>> => {
+    const signatureResp = await this.getLoginSignature(captchaToken)
+    if (!signatureResp.ok) {
+      return signatureResp
+    }
+
+    const payload: StartSessionPayload = {
+      externalAccount: this.externalAccount,
+      captchaResponseToken: captchaToken,
+      signature: signatureResp.result,
+    }
+
+    const resp = await this.client.exec(startSession(payload))
+
+    if (resp.ok) {
+      this.client.setToken(resp.result.token)
+      return Ok(resp.result.token)
+    } else if (resp.error.errorType === FetchErrorTypes.Unauthorised) {
+      return Err(new AuthenticationFailed())
+    }
+
+    return resp
+  }
+
+  /**
+   * getDistributedBlindedPepper: uses the /v1/distributedBlindedPepper endpoint to
+   * request the identifier and pepper associated with a phone number in a fee-less scenario
+   *
+   * @param e164Number - phone number
+   * @param clientVersion
+   * @returns the identifier and the pepper
+   */
+  getDistributedBlindedPepper = async (
+    e164Number: string,
+    clientVersion: string
```

Comment:
@codyborn, keep me honest but AFAIK client version is the actual version of Valora, and it's something that Odis requires for tracking I guess. It's currently passed in as `DeviceInfo.getVersion()` in `identity/privateHashing.ts`.

Reply:
That's right. It's purely for monitoring and lets us know on the ODIS side how many users are still using a given version. This can be helpful when deprecating old code paths.